lms/djangoapps/instructor/views/data_access.py

```diff
@@ -0,0 +1,374 @@
+    """
+    if query.filter == SectionFilters.OPENED:
+        results = open_query(course_id, query)
+    elif query.filter == SectionFilters.NOT_OPENED:
+        results = not_open_query(course_id, query)
+    return results
+
+
+def make_existing_query(existing_queries):
+    """
+    Aggregates single queries in a group into one unified set of students
+    """
+    query = User.objects
+    if existing_queries is None:
+        return None
+
+    query_dct = defaultdict(list)
+
+    for existing_query in existing_queries:
+        if existing_query == "" or existing_query == QueryStatus.WORKING:
+            continue
+        inclusion_type = TemporaryQuery.objects.filter(id=existing_query)
+        result = StudentsForQuery.objects.filter(query_id=existing_query).values_list(
+            DatabaseFields.STUDENT_ID,
+            flat=True,
+        )
+        if inclusion_type.exists():
+            query_dct[inclusion_type[0].inclusion].append(result)
+
+    for not_query in query_dct[INCLUSION_MAP.get(Inclusion.NOT)]:
+        query = query.exclude(id__in=not_query)
+
+    for and_query in query_dct[INCLUSION_MAP.get(Inclusion.AND)]:
+        query = query.filter(id__in=and_query)
+
+    or_query = User.objects
+    qobjs = Q()
+    for orq in query_dct[INCLUSION_MAP.get(Inclusion.OR)]:
+        qobjs = qobjs | (Q(id__in=orq))
+    if len(query_dct[INCLUSION_MAP.get(Inclusion.NOT)]) == 0 and len(query_dct[INCLUSION_MAP.get(Inclusion.AND)]) == 0:
+        return or_query.filter(qobjs)
+    else:
+        return query | or_query.filter(qobjs)
+
+
+def not_open_query(course_id, query):
+    """
+    Specific db query for sections or problems that have not been opened
+    """
+    ids_in_course = CourseEnrollment.objects.filter(course_id=course_id,
+                                                    is_active=1,
+                                                    ).values_list(DatabaseFields.USER_ID)
+    total_students = User.objects.filter(id__in=ids_in_course)
+    without_open = total_students.exclude(id__in=
+                                          StudentModule.objects.filter(
+                                              module_state_key=query.entity_id,
+                                              course_id=course_id).values_list(DatabaseFields.STUDENT_ID),
+                                          )
+    return process_results(course_id, without_open, DatabaseFields.ID, DatabaseFields.EMAIL)
+
+
+def not_completed_query(course_id, query):
+    """
+    Specific db query for sections or problems that are not completed
+    """
+    ids_in_course = CourseEnrollment.objects.filter(course_id=course_id,
+                                                    is_active=1,
+                                                    ).values_list(DatabaseFields.USER_ID)
+    total_students = User.objects.filter(id__in=ids_in_course)
+    without_completed = total_students.exclude(id__in=
```

Comment:
Found out that exclude is different than filter because if a student has never opened the problem, they would not be in the studentmodule table and thus filtering would not be the right thing to do